src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java

```diff
@@ -72,22 +70,26 @@ private static void setupDrivers(Application application) {
 		DriverFeature.addDriver("Logger driver", loggerDriver);
 
 		DrawPanelController drawerController = DrawerFeature.getDrawerController();
-		Job2dDriver driver = new LineDriverAdapter(drawerController, LineFactory.getBasicLine(), "basic");
-		DriverFeature.addDriver("Line Simulator", driver);
-		DriverFeature.getDriverManager().setCurrentDriver(driver);
+		Job2dDriver basicLineDriver = new LineDriverAdapter(drawerController, LineFactory.getBasicLine(), "basic");
+		DriverFeature.addDriver("Line Simulator", basicLineDriver);
+		DriverFeature.getDriverManager().setCurrentDriver(basicLineDriver);
+
+		Job2dDriver specialLineDriver = new LineDriverAdapter(drawerController, LineFactory.getSpecialLine(), "special");
+		DriverFeature.addDriver("Special line Simulator", specialLineDriver);
 
-		driver = new LineDriverAdapter(drawerController, LineFactory.getSpecialLine(), "special");
-		DriverFeature.addDriver("Special line Simulator", driver);
+		Job2dDriver dottedLineDriver = new LineDriverAdapter(drawerController, LineFactory.getDottedLine(), "dotted");
+		DriverFeature.addDriver("Dotted line Simulator", dottedLineDriver);
 
-		driver = new LineDriverAdapter(drawerController, LineFactory.getDottedLine(), "dotted");
-		DriverFeature.addDriver("Dotted line Simulator", driver);
+		DriverFeature.addDriver("Start Macro Driver", MacroFeature.getMacroDriver());
 
-		DriverFeature.addDriver("Start Macro Driver", MacroFeature.getMacroDriverDecorator());
-		MacroFeature.getMacroDriverDecorator().setCoreJob2dDriver(driver);
+		DriverComposite driverComposite = new DriverComposite();
+		driverComposite.addDriver(basicLineDriver);
+		MacroFeature.getMacroDriver().setDriverComposite(driverComposite);
```

Comment:
No like this! The other way around:
driverComposite.addDriver(MacroFeature.getMacroDriver());
Is it clear now? ;)